parse.go

```diff
@@ -46,199 +33,8 @@ func ParseEncryptedPrivateKey(data []byte, passphrase []byte) (ssh.Signer, error
 // but some formats like RSA in PKCS#1 detecting a wrong passphrase is difficult,
 // and other parse errors may be returned.
 func ParseEncryptedRawPrivateKey(data []byte, passphrase []byte) (interface{}, error) {
-	var err error
-
-	block, _ := pem.Decode(data)
-	if block == nil {
-		return nil, errors.New("no PEM block found")
-	}
-
-	if x509.IsEncryptedPEMBlock(block) {
-		data, err = x509.DecryptPEMBlock(block, passphrase)
-		if err == x509.IncorrectPasswordError {
-			return nil, ErrIncorrectPassword
-		}
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		data = block.Bytes
-	}
-
-	switch block.Type {
-	case "RSA PRIVATE KEY":
-		pk, err := x509.ParsePKCS1PrivateKey(data)
-		if err != nil {
-			// The Algos for PEM Encryption do not include strong message authentication,
-			// so sometimes DecryptPEMBlock works, but ParsePKCS1PrivateKey fails with an asn1 error.
-			// We are just catching the most common prefix here...
-			if strings.HasPrefix(err.Error(), "asn1: structure error") {
-				return nil, ErrIncorrectPassword
```

Comment:
this is one thing that they didn't do upstream, pretty minor, and I guess it's ok to sometimes return an asn1 parse error.

Reply:
Tests didn't hit it :-)